src/beatree/writeout.rs

```diff
@@ -264,18 +281,62 @@ impl BbnWriteOut {
 struct LnWriteOut {
     ln_fd: RawFd,
     ln_extend_file_sz: Option<u64>,
-    ln: Vec<()>,
+    ln: Vec<(PageNumber, Box<Page>)>,
+    ln_remaining: usize,
 }
 
 impl LnWriteOut {
-    fn run(&mut self) -> bool {
+    fn run(&mut self, io: &mut IoDmux) -> bool {
+        // NOTE: a possible improvement could be to allow the LeafStore to perform write operations
+        // directly into itself if it falls within an already allocated file.
+        // The `writeout` function would then only need to call ftruncate and write any exceeded pages.
```

Comment:
1. It's not clear what **exactly** is suggested. 
2. If I understand it right, then I am if I agree in what way this would be an improvement.

Reply:
I will make it more clear!